chia/farmer/farmer.py

```diff
@@ -183,30 +185,67 @@ async def setup_keys(self):
         self.harvester_cache: Dict[str, Dict[str, HarvesterCacheEntry]] = {}
 
     async def _start(self):
-        await self.setup_keys()
-        self.update_pool_state_task = asyncio.create_task(self._periodically_update_pool_state_task())
-        self.cache_clear_task = asyncio.create_task(self._periodically_clear_cache_and_refresh_task())
+        async def start_task():
+            # `Farmer.setup_keys` throws if there are no keys setup yet. In this case we just try until it succeeds or
+            # until we need to shut down.
+            while not self._shut_down:
+                try:
+                    await self.setup_keys()
+                except Exception as e:
```

Comment:
Is there a more specific exception we should be catching that identifies the case where keys aren't setup yet? Otherwise, an unrelated exception may cause this loop to never exit.

Reply:
Good point - one could argue it should throw a chia subclass kind of exception so we know exactly what to catch. I think now it throws RuntimeError with a specific string.

Reply:
I just dropped that exception, see b163295.